// @didit-protocol/sdk-web/dist/didit-sdk.umd.js
/**
 * Didit SDK for Web v0.1.9
 * (c) 2026 Didit
 * @license MIT
 */
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define(['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.DiditSDK = {}));
})(this, (function (exports) { 'use strict';

    const DEFAULT_CONFIG = {
        zIndex: 9999,
        showCloseButton: true,
        showExitConfirmation: true,
        loggingEnabled: false};
    const CSS_CLASSES = {
        overlay: "didit-modal-overlay",
        container: "didit-modal-container",
        iframe: "didit-verification-iframe",
        closeButton: "didit-close-button",
        loading: "didit-loading",
        confirmOverlay: "didit-confirm-overlay",
        confirmBox: "didit-confirm-box",
        embedded: "didit-embedded"
    };
    const SDK_VERSION = "0.1.9";
    const languages = [
        "ar",
        "bg",
        "bn",
        "ca",
        "cnr",
        "cs",
        "da",
        "de",
        "el",
        "en",
        "es",
        "et",
        "fa",
        "fi",
        "fr",
        "he",
        "hi",
        "hr",
        "hu",
        "hy",
        "id",
        "it",
        "ja",
        "ka",
        "ko",
        "lt",
        "lv",
        "mk",
        "ms",
        "nl",
        "no",
        "pl",
        "pt-BR",
        "pt",
        "ro",
        "ru",
        "sk",
        "sl",
        "so",
        "sr",
        "sv",
        "th",
        "tr",
        "uk",
        "uz",
        "vi",
        "zh-CN",
        "zh-TW",
        "zh",
    ];

    class SDKLogger {
        static get isEnabled() {
            return this._enabled;
        }
        static set isEnabled(value) {
            this._enabled = value;
        }
        static log(...args) {
            if (this._enabled) {
                console.log("[DiditSDK]", ...args);
            }
        }
        static warn(...args) {
            if (this._enabled) {
                console.warn("[DiditSDK]", ...args);
            }
        }
        static error(...args) {
            if (this._enabled) {
                console.error("[DiditSDK]", ...args);
            }
        }
    }
    SDKLogger._enabled = false;
    function generateModalId() {
        return `didit-modal-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;
    }
    function isAllowedOrigin(origin) {
        try {
            const url = new URL(origin);
            return url.hostname.endsWith(".didit.me");
        }
        catch {
            return false;
        }
    }
    function createVerificationError(type, customMessage) {
        const messages = {
            sessionExpired: "Your verification session has expired.",
            networkError: "A network error occurred. Please try again.",
            cameraAccessDenied: "Camera access is required for verification.",
            unknown: customMessage || "An unknown error occurred."
        };
        return {
            type,
            message: customMessage || messages[type]
        };
    }
    function detectLanguageFromUrl(url) {
        try {
            const { pathname } = new URL(url);
            const firstSegment = pathname.split("/").filter(Boolean)[0];
            if (firstSegment && languages.includes(firstSegment)) {
                return firstSegment;
            }
        }
        catch {
            // we get it from the browser
        }
        const browserLang = navigator.language;
        if (languages.includes(browserLang))
            return browserLang;
        const baseLang = browserLang.split("-")[0];
        if (languages.includes(baseLang))
            return baseLang;
        return "en";
    }

    const EN = {
        exitTitle: "Exit verification?",
        exitMessage: "Exiting will end your verification process. Are you sure?",
        continueButton: "Continue",
        exitButton: "Exit",
        ariaLabelModal: "Didit Verification",
        ariaLabelClose: "Close verification"
    };
    const translationsMap = {
        ar: {
            exitTitle: "الخروج من التحقق؟",
            exitMessage: "سيؤدي الخروج إلى إنهاء عملية التحقق الخاصة بك. هل أنت متأكد؟",
            continueButton: "متابعة",
            exitButton: "خروج",
            ariaLabelModal: "التحقق من Didit",
            ariaLabelClose: "إغلاق التحقق"
        },
        bg: {
            exitTitle: "Излизане от верификацията?",
            exitMessage: "Излизането ще прекрати процеса на верификация. Сигурни ли сте?",
            continueButton: "Продължи",
            exitButton: "Изход",
            ariaLabelModal: "Верификация Didit",
            ariaLabelClose: "Затваряне на верификацията"
        },
        bn: {
            exitTitle: "যাচাইকরণ থেকে বের হবেন?",
            exitMessage: "বের হলে আপনার যাচাইকরণ প্রক্রিয়া শেষ হয়ে যাবে। আপনি কি নিশ্চিত?",
            continueButton: "চালিয়ে যান",
            exitButton: "বের হন",
            ariaLabelModal: "Didit যাচাইকরণ",
            ariaLabelClose: "যাচাইকরণ বন্ধ করুন"
        },
        ca: {
            exitTitle: "Sortir de la verificació?",
            exitMessage: "Sortir finalitzarà el procés de verificació. N'esteu segur?",
            continueButton: "Continua",
            exitButton: "Sortir",
            ariaLabelModal: "Verificació Didit",
            ariaLabelClose: "Tancar verificació"
        },
        cnr: {
            exitTitle: "Izaći iz verifikacije?",
            exitMessage: "Izlaskom ćete prekinuti proces verifikacije. Jeste li sigurni?",
            continueButton: "Nastavi",
            exitButton: "Izađi",
            ariaLabelModal: "Didit verifikacija",
            ariaLabelClose: "Zatvori verifikaciju"
        },
        cs: {
            exitTitle: "Opravdu chcete odejít?",
            exitMessage: "Když odejdete, proces ověření se přeruší. Opravdu chcete skončit?",
            continueButton: "Zůstat",
            exitButton: "Odejít",
            ariaLabelModal: "Ověření Didit",
            ariaLabelClose: "Zavřít ověření"
        },
        da: {
            exitTitle: "Forlad verifikation?",
            exitMessage: "Hvis du forlader, afsluttes din verifikationsproces. Er du sikker?",
            continueButton: "Fortsæt",
            exitButton: "Forlad",
            ariaLabelModal: "Didit-verifikation",
            ariaLabelClose: "Luk verifikation"
        },
        de: {
            exitTitle: "Verifizierung verlassen?",
            exitMessage: "Das Verlassen beendet Ihren Verifizierungsprozess. Sind Sie sicher?",
            continueButton: "Fortfahren",
            exitButton: "Verlassen",
            ariaLabelModal: "Didit-Verifizierung",
            ariaLabelClose: "Verifizierung schließen"
        },
        el: {
            exitTitle: "Έξοδος από την επαλήθευση;",
            exitMessage: "Η έξοδος θα τερματίσει τη διαδικασία επαλήθευσης. Είστε σίγουροι;",
            continueButton: "Συνέχεια",
            exitButton: "Έξοδος",
            ariaLabelModal: "Επαλήθευση Didit",
            ariaLabelClose: "Κλείσιμο επαλήθευσης"
        },
        en: EN,
        es: {
            exitTitle: "¿Salir de la verificación?",
            exitMessage: "Salir terminará tu proceso de verificación. ¿Estás seguro?",
            continueButton: "Continuar",
            exitButton: "Salir",
            ariaLabelModal: "Verificación Didit",
            ariaLabelClose: "Cerrar verificación"
        },
        et: {
            exitTitle: "Lahkuda kinnitamisest?",
            exitMessage: "Lahkumine lõpetab teie kinnitamisprotsessi. Kas olete kindel?",
            continueButton: "Jätka",
            exitButton: "Lahku",
            ariaLabelModal: "Didit kinnitus",
            ariaLabelClose: "Sulge kinnitus"
        },
        fa: {
            exitTitle: "خروج از تأیید هویت؟",
            exitMessage: "خروج باعث پایان فرآیند تأیید هویت شما می‌شود. آیا مطمئن هستید؟",
            continueButton: "ادامه",
            exitButton: "خروج",
            ariaLabelModal: "تأیید هویت Didit",
            ariaLabelClose: "بستن تأیید هویت"
        },
        fi: {
            exitTitle: "Poistu vahvistuksesta?",
            exitMessage: "Poistuminen päättää vahvistusprosessisi. Oletko varma?",
            continueButton: "Jatka",
            exitButton: "Poistu",
            ariaLabelModal: "Didit-vahvistus",
            ariaLabelClose: "Sulje vahvistus"
        },
        fr: {
            exitTitle: "Quitter la vérification ?",
            exitMessage: "Quitter mettra fin à votre processus de vérification. Êtes-vous sûr ?",
            continueButton: "Continuer",
            exitButton: "Quitter",
            ariaLabelModal: "Vérification Didit",
            ariaLabelClose: "Fermer la vérification"
        },
        he: {
            exitTitle: "לצאת מהאימות?",
            exitMessage: "יציאה תסיים את תהליך האימות שלך. האם אתה בטוח?",
            continueButton: "המשך",
            exitButton: "יציאה",
            ariaLabelModal: "אימות Didit",
            ariaLabelClose: "סגירת אימות"
        },
        hi: {
            exitTitle: "सत्यापन से बाहर निकलें?",
            exitMessage: "बाहर निकलने से आपकी सत्यापन प्रक्रिया समाप्त हो जाएगी। क्या आप सुनिश्चित हैं?",
            continueButton: "जारी रखें",
            exitButton: "बाहर निकलें",
            ariaLabelModal: "Didit सत्यापन",
            ariaLabelClose: "सत्यापन बंद करें"
        },
        hr: {
            exitTitle: "Izaći iz verifikacije?",
            exitMessage: "Izlaskom ćete prekinuti proces verifikacije. Jeste li sigurni?",
            continueButton: "Nastavi",
            exitButton: "Izađi",
            ariaLabelModal: "Didit verifikacija",
            ariaLabelClose: "Zatvori verifikaciju"
        },
        hu: {
            exitTitle: "Kilépés az ellenőrzésből?",
            exitMessage: "A kilépés befejezi az ellenőrzési folyamatot. Biztos benne?",
            continueButton: "Folytatás",
            exitButton: "Kilépés",
            ariaLabelModal: "Didit ellenőrzés",
            ariaLabelClose: "Ellenőrzés bezárása"
        },
        hy: {
            exitTitle: "Դուրս գա՞լ ստուգումից",
            exitMessage: "Դուրս գալը կավարտի ձեր ստուգման գործընթացը։ Համոզված ե՞ք?",
            continueButton: "Շարունակել",
            exitButton: "Դուրս գալ",
            ariaLabelModal: "Didit ստուգում",
            ariaLabelClose: "Փակել ստուգումը"
        },
        id: {
            exitTitle: "Keluar dari verifikasi?",
            exitMessage: "Keluar akan mengakhiri proses verifikasi Anda. Apakah Anda yakin?",
            continueButton: "Lanjutkan",
            exitButton: "Keluar",
            ariaLabelModal: "Verifikasi Didit",
            ariaLabelClose: "Tutup verifikasi"
        },
        it: {
            exitTitle: "Uscire dalla verifica?",
            exitMessage: "L'uscita terminerà il processo di verifica. Sei sicuro?",
            continueButton: "Continua",
            exitButton: "Esci",
            ariaLabelModal: "Verifica Didit",
            ariaLabelClose: "Chiudi verifica"
        },
        ja: {
            exitTitle: "認証を終了しますか？",
            exitMessage: "終了すると認証プロセスが中断されます。よろしいですか？",
            continueButton: "続ける",
            exitButton: "終了",
            ariaLabelModal: "Didit 認証",
            ariaLabelClose: "認証を閉じる"
        },
        ka: {
            exitTitle: "გამოსვლა შემოწმებიდან?",
            exitMessage: "გამოსვლა დაასრულებს თქვენს შემოწმების პროცესს. დარწმუნებული ხართ?",
            continueButton: "გაგრძელება",
            exitButton: "გამოსვლა",
            ariaLabelModal: "Didit შემოწმება",
            ariaLabelClose: "შემოწმების დახურვა"
        },
        ko: {
            exitTitle: "인증을 종료하시겠습니까?",
            exitMessage: "종료하면 인증 절차가 중단됩니다. 확실하십니까?",
            continueButton: "계속",
            exitButton: "종료",
            ariaLabelModal: "Didit 인증",
            ariaLabelClose: "인증 닫기"
        },
        lt: {
            exitTitle: "Išeiti iš patvirtinimo?",
            exitMessage: "Išėjimas nutrauks jūsų patvirtinimo procesą. Ar esate tikri?",
            continueButton: "Tęsti",
            exitButton: "Išeiti",
            ariaLabelModal: "Didit patvirtinimas",
            ariaLabelClose: "Uždaryti patvirtinimą"
        },
        lv: {
            exitTitle: "Iziet no verifikācijas?",
            exitMessage: "Iziešana pārtrauks jūsu verifikācijas procesu. Vai esat pārliecināts?",
            continueButton: "Turpināt",
            exitButton: "Iziet",
            ariaLabelModal: "Didit verifikācija",
            ariaLabelClose: "Aizvērt verifikāciju"
        },
        mk: {
            exitTitle: "Излези од верификацијата?",
            exitMessage: "Излегувањето ќе го прекине процесот на верификација. Дали сте сигурни?",
            continueButton: "Продолжи",
            exitButton: "Излези",
            ariaLabelModal: "Верификација Didit",
            ariaLabelClose: "Затвори верификација"
        },
        ms: {
            exitTitle: "Keluar dari pengesahan?",
            exitMessage: "Keluar akan menamatkan proses pengesahan anda. Adakah anda pasti?",
            continueButton: "Teruskan",
            exitButton: "Keluar",
            ariaLabelModal: "Pengesahan Didit",
            ariaLabelClose: "Tutup pengesahan"
        },
        nl: {
            exitTitle: "Verificatie verlaten?",
            exitMessage: "Verlaten beëindigt uw verificatieproces. Weet u het zeker?",
            continueButton: "Doorgaan",
            exitButton: "Verlaten",
            ariaLabelModal: "Didit-verificatie",
            ariaLabelClose: "Verificatie sluiten"
        },
        no: {
            exitTitle: "Forlat verifisering?",
            exitMessage: "Å forlate vil avslutte verifiseringsprosessen. Er du sikker?",
            continueButton: "Fortsett",
            exitButton: "Forlat",
            ariaLabelModal: "Didit-verifisering",
            ariaLabelClose: "Lukk verifisering"
        },
        pl: {
            exitTitle: "Czy wyjść z weryfikacji?",
            exitMessage: "Wyjście zakończy proces weryfikacji. Czy na pewno?",
            continueButton: "Kontynuuj",
            exitButton: "Wyjdź",
            ariaLabelModal: "Weryfikacja Didit",
            ariaLabelClose: "Zamknij weryfikację"
        },
        "pt-BR": {
            exitTitle: "Sair da verificação?",
            exitMessage: "Sair encerrará seu processo de verificação. Tem certeza?",
            continueButton: "Continuar",
            exitButton: "Sair",
            ariaLabelModal: "Verificação Didit",
            ariaLabelClose: "Fechar verificação"
        },
        pt: {
            exitTitle: "Sair da verificação?",
            exitMessage: "Sair terminará o seu processo de verificação. Tem a certeza?",
            continueButton: "Continuar",
            exitButton: "Sair",
            ariaLabelModal: "Verificação Didit",
            ariaLabelClose: "Fechar verificação"
        },
        ro: {
            exitTitle: "Ieși din verificare?",
            exitMessage: "Ieșirea va încheia procesul de verificare. Ești sigur?",
            continueButton: "Continuă",
            exitButton: "Ieși",
            ariaLabelModal: "Verificare Didit",
            ariaLabelClose: "Închide verificarea"
        },
        ru: {
            exitTitle: "Выйти из верификации?",
            exitMessage: "Выход завершит процесс верификации. Вы уверены?",
            continueButton: "Продолжить",
            exitButton: "Выйти",
            ariaLabelModal: "Верификация Didit",
            ariaLabelClose: "Закрыть верификацию"
        },
        sk: {
            exitTitle: "Opustiť overenie?",
            exitMessage: "Odchodom ukončíte proces overenia. Ste si istí?",
            continueButton: "Pokračovať",
            exitButton: "Odísť",
            ariaLabelModal: "Overenie Didit",
            ariaLabelClose: "Zavrieť overenie"
        },
        sl: {
            exitTitle: "Zapustiti preverjanje?",
            exitMessage: "Izhod bo prekinil postopek preverjanja. Ali ste prepričani?",
            continueButton: "Nadaljuj",
            exitButton: "Izhod",
            ariaLabelModal: "Preverjanje Didit",
            ariaLabelClose: "Zapri preverjanje"
        },
        so: {
            exitTitle: "Ka baxdo xaqiijinta?",
            exitMessage: "Ka bixitaanku wuxuu dhammayn doonaa habka xaqiijintaada. Ma hubtaa?",
            continueButton: "Sii wad",
            exitButton: "Ka bax",
            ariaLabelModal: "Xaqiijinta Didit",
            ariaLabelClose: "Xir xaqiijinta"
        },
        sr: {
            exitTitle: "Изаћи из верификације?",
            exitMessage: "Изласком ћете прекинути процес верификације. Да ли сте сигурни?",
            continueButton: "Настави",
            exitButton: "Изађи",
            ariaLabelModal: "Верификација Didit",
            ariaLabelClose: "Затвори верификацију"
        },
        sv: {
            exitTitle: "Lämna verifiering?",
            exitMessage: "Att lämna avslutar din verifieringsprocess. Är du säker?",
            continueButton: "Fortsätt",
            exitButton: "Lämna",
            ariaLabelModal: "Didit-verifiering",
            ariaLabelClose: "Stäng verifiering"
        },
        th: {
            exitTitle: "ออกจากการยืนยันตัวตน?",
            exitMessage: "การออกจะสิ้นสุดกระบวนการยืนยันตัวตนของคุณ คุณแน่ใจหรือไม่?",
            continueButton: "ดำเนินการต่อ",
            exitButton: "ออก",
            ariaLabelModal: "การยืนยันตัวตน Didit",
            ariaLabelClose: "ปิดการยืนยันตัวตน"
        },
        tr: {
            exitTitle: "Doğrulamadan çıkmak istiyor musunuz?",
            exitMessage: "Çıkış, doğrulama sürecinizi sonlandıracak. Emin misiniz?",
            continueButton: "Devam et",
            exitButton: "Çıkış",
            ariaLabelModal: "Didit doğrulama",
            ariaLabelClose: "Doğrulamayı kapat"
        },
        uk: {
            exitTitle: "Вийти з верифікації?",
            exitMessage: "Вихід завершить процес верифікації. Ви впевнені?",
            continueButton: "Продовжити",
            exitButton: "Вийти",
            ariaLabelModal: "Верифікація Didit",
            ariaLabelClose: "Закрити верифікацію"
        },
        uz: {
            exitTitle: "Tekshiruvdan chiqasizmi?",
            exitMessage: "Chiqish tekshiruv jarayonini tugatadi. Ishonchingiz komilmi?",
            continueButton: "Davom etish",
            exitButton: "Chiqish",
            ariaLabelModal: "Didit tekshiruvi",
            ariaLabelClose: "Tekshiruvni yopish"
        },
        vi: {
            exitTitle: "Thoát khỏi xác minh?",
            exitMessage: "Thoát sẽ kết thúc quá trình xác minh của bạn. Bạn có chắc không?",
            continueButton: "Tiếp tục",
            exitButton: "Thoát",
            ariaLabelModal: "Xác minh Didit",
            ariaLabelClose: "Đóng xác minh"
        },
        "zh-CN": {
            exitTitle: "退出验证？",
            exitMessage: "退出将结束您的验证流程。确定要退出吗？",
            continueButton: "继续",
            exitButton: "退出",
            ariaLabelModal: "Didit 验证",
            ariaLabelClose: "关闭验证"
        },
        "zh-TW": {
            exitTitle: "退出驗證？",
            exitMessage: "退出將結束您的驗證流程。確定要退出嗎？",
            continueButton: "繼續",
            exitButton: "退出",
            ariaLabelModal: "Didit 驗證",
            ariaLabelClose: "關閉驗證"
        },
        zh: {
            exitTitle: "退出验证？",
            exitMessage: "退出将结束您的验证流程。确定要退出吗？",
            continueButton: "继续",
            exitButton: "退出",
            ariaLabelModal: "Didit 验证",
            ariaLabelClose: "关闭验证"
        }
    };
    function getTranslations(language) {
        return translationsMap[language] ?? EN;
    }

    class VerificationModal {
        constructor(configuration, callbacks) {
            this.state = {
                isOpen: false,
                isLoading: true,
                showConfirmation: false
            };
            this.overlay = null;
            this.container = null;
            this.iframe = null;
            this.loadingEl = null;
            this.confirmOverlay = null;
            this.boundHandleMessage = null;
            this.boundHandleKeydown = null;
            this.embedded = false;
            this.embeddedContainer = null;
            this.language = "en";
            this.modalId = generateModalId();
            this.config = {
                zIndex: configuration?.zIndex ?? DEFAULT_CONFIG.zIndex,
                showCloseButton: configuration?.showCloseButton ?? DEFAULT_CONFIG.showCloseButton,
                showExitConfirmation: configuration?.showExitConfirmation ?? DEFAULT_CONFIG.showExitConfirmation
            };
            this.callbacks = callbacks;
            this.containerElement = configuration?.containerElement ?? document.body;
            this.embedded = configuration?.embedded ?? false;
            if (this.embedded && configuration?.embeddedContainerId) {
                this.embeddedContainer = document.getElementById(configuration.embeddedContainerId);
            }
        }
        injectStyles() {
            const styleId = "didit-sdk-styles";
            if (document.getElementById(styleId)) {
                return;
            }
            const styles = document.createElement("style");
            styles.id = styleId;
            styles.textContent = `
      .${CSS_CLASSES.overlay} {
        display: none;
        position: fixed;
        inset: 0;
        background: rgba(0, 0, 0, 0.6);
        z-index: ${this.config.zIndex};
        justify-content: center;
        align-items: center;
        padding: 1rem;
        opacity: 0;
        transition: opacity 0.2s ease-out;
      }

      .${CSS_CLASSES.overlay}.active {
        display: flex;
        opacity: 1;
      }

      .${CSS_CLASSES.container} {
        position: relative;
        width: 100%;
        max-width: 500px;
        max-height: 90dvh;
        border-radius: 16px;
        overflow: hidden;
        background: transparent;
      }

      .${CSS_CLASSES.overlay}.active .${CSS_CLASSES.container} {
        transform: scale(1);
      }

      .${CSS_CLASSES.iframe} {
        width: 100%;
        height: 700px;
        border: none;
        display: block;
      }

      .${CSS_CLASSES.closeButton} {
        position: absolute;
        top: 4px;
        right: 4px;
        width: 24px;
        height: 24px;
        background: transparent;
        border: none;
        cursor: pointer;
        padding: 0;
        z-index: 10;
        outline: none;
      }

      .${CSS_CLASSES.closeButton}:hover,
      .${CSS_CLASSES.closeButton}:focus {
        background: transparent;
        opacity: 0.5;
      }

      .${CSS_CLASSES.closeButton} svg {
        stroke: #666;
        stroke-width: 2;
        stroke-linecap: round;
      }

      .${CSS_CLASSES.loading} {
        position: absolute;
        inset: 0;
        display: flex;
        align-items: center;
        justify-content: center;
        background: #fafafa;
        z-index: 5;
      }

      .${CSS_CLASSES.loading}.hidden {
        display: none;
      }

      .${CSS_CLASSES.loading} svg {
        width: 4rem;
        height: 4rem;
        animation: didit-spin 1s linear infinite;
      }

      .${CSS_CLASSES.loading} circle {
        stroke: #e5e5e5;
        stroke-width: 2.5;
        fill: none;
      }

      .${CSS_CLASSES.loading} path {
        stroke: #525252;
        stroke-width: 2.5;
        stroke-linecap: round;
        fill: none;
      }

      @keyframes didit-spin {
        from { transform: rotate(0deg); }
        to { transform: rotate(360deg); }
      }

      .${CSS_CLASSES.confirmOverlay} {
        display: none;
        position: absolute;
        inset: 0;
        background: rgba(0, 0, 0, 0.5);
        z-index: 20;
        justify-content: center;
        align-items: center;
        opacity: 0;
        transition: opacity 0.15s ease-out;
      }

      .${CSS_CLASSES.confirmOverlay}.active {
        display: flex;
        opacity: 1;
      }

      .${CSS_CLASSES.confirmBox} {
        background: #fff;
        border-radius: 12px;
        padding: 1.5rem;
        text-align: center;
        max-width: 300px;
        margin: 1rem;
        transform: scale(0.95);
        transition: transform 0.15s ease-out;
        box-shadow: 0 4px 20px rgba(0, 0, 0, 0.15);
      }

      .${CSS_CLASSES.confirmOverlay}.active .${CSS_CLASSES.confirmBox} {
        transform: scale(1);
      }

      .${CSS_CLASSES.confirmBox} h3 {
        color: #1a1a2e;
        margin: 0 0 0.5rem 0;
        font-size: 1.125rem;
        font-weight: 600;
      }

      .${CSS_CLASSES.confirmBox} p {
        color: #666;
        font-size: 0.875rem;
        margin: 0 0 1.25rem 0;
        line-height: 1.5;
      }

      .didit-confirm-actions {
        display: flex;
        align-items: center;
        justify-content: center;
        gap: 1rem;
      }

      .didit-confirm-actions button {
        background: #2563eb;
        color: #fff;
        border: none;
        padding: 0.625rem 1.25rem;
        border-radius: 8px;
        font-size: 0.875rem;
        font-weight: 500;
        cursor: pointer;
        transition: background 0.15s ease;
      }

      .didit-confirm-actions button:hover {
        background: #1d4ed8;
      }

      .didit-confirm-actions span {
        color: #666;
        font-size: 0.875rem;
        cursor: pointer;
        padding: 0.625rem;
        transition: color 0.15s ease;
      }

      .didit-confirm-actions span:hover {
        color: #1a1a2e;
      }

      @media (max-width: 540px) {
        .${CSS_CLASSES.overlay} {
          padding: 0;
        }

        .${CSS_CLASSES.container} {
          max-width: 100%;
          max-height: 100dvh;
          border-radius: 0;
        }

        .${CSS_CLASSES.iframe} {
          height: 100dvh;
        }
      }

      .${CSS_CLASSES.embedded} {
        position: relative;
        width: 100%;
        height: 100%;
      }

      .${CSS_CLASSES.embedded} .${CSS_CLASSES.iframe} {
        width: 100%;
        height: 100%;
      }

      .${CSS_CLASSES.embedded} .${CSS_CLASSES.loading} {
        border-radius: 0;
      }
    `;
            document.head.appendChild(styles);
        }
        createDOM() {
            this.injectStyles();
            if (this.embedded && this.embeddedContainer) {
                this.createEmbeddedDOM();
                return;
            }
            const t = getTranslations(this.language);
            this.overlay = document.createElement("div");
            this.overlay.id = this.modalId;
            this.overlay.className = CSS_CLASSES.overlay;
            this.overlay.setAttribute("role", "dialog");
            this.overlay.setAttribute("aria-modal", "true");
            this.overlay.setAttribute("aria-label", t.ariaLabelModal);
            this.container = document.createElement("div");
            this.container.className = CSS_CLASSES.container;
            this.loadingEl = document.createElement("div");
            this.loadingEl.className = CSS_CLASSES.loading;
            this.loadingEl.innerHTML = `
      <svg viewBox="0 0 24 24">
        <circle cx="12" cy="12" r="10" />
        <path d="M12 2a10 10 0 0 1 10 10" />
      </svg>
    `;
            if (this.config.showCloseButton) {
                const closeBtn = document.createElement("button");
                closeBtn.className = CSS_CLASSES.closeButton;
                closeBtn.setAttribute("aria-label", t.ariaLabelClose);
                closeBtn.innerHTML = `
        <svg width="14" height="14" viewBox="0 0 14 14">
          <line x1="1" y1="1" x2="13" y2="13" />
          <line x1="13" y1="1" x2="1" y2="13" />
        </svg>
      `;
                closeBtn.addEventListener("click", () => this.handleCloseRequest());
                this.container.appendChild(closeBtn);
            }
            this.iframe = document.createElement("iframe");
            this.iframe.className = CSS_CLASSES.iframe;
            this.iframe.setAttribute("allow", "camera; microphone; fullscreen; autoplay; encrypted-media; geolocation");
            this.iframe.setAttribute("title", t.ariaLabelModal);
            this.iframe.addEventListener("load", () => this.handleIframeLoad());
            this.confirmOverlay = document.createElement("div");
            this.confirmOverlay.className = CSS_CLASSES.confirmOverlay;
            this.confirmOverlay.innerHTML = `
      <div class="${CSS_CLASSES.confirmBox}">
        <h3>${t.exitTitle}</h3>
        <p>${t.exitMessage}</p>
        <div class="didit-confirm-actions">
          <button type="button" data-action="continue">${t.continueButton}</button>
          <span data-action="exit">${t.exitButton}</span>
        </div>
      </div>
    `;
            this.confirmOverlay.querySelector('[data-action="continue"]')?.addEventListener("click", () => {
                this.hideConfirmation();
            });
            this.confirmOverlay.querySelector('[data-action="exit"]')?.addEventListener("click", () => {
                this.confirmExit();
            });
            this.container.appendChild(this.loadingEl);
            this.container.appendChild(this.iframe);
            this.container.appendChild(this.confirmOverlay);
            this.overlay.appendChild(this.container);
            this.overlay.addEventListener("click", (e) => {
                if (e.target === this.overlay) {
                    this.handleCloseRequest();
                }
            });
            this.containerElement.appendChild(this.overlay);
        }
        createEmbeddedDOM() {
            if (!this.embeddedContainer)
                return;
            this.container = document.createElement("div");
            this.container.id = this.modalId;
            this.container.className = CSS_CLASSES.embedded;
            this.loadingEl = document.createElement("div");
            this.loadingEl.className = CSS_CLASSES.loading;
            this.loadingEl.innerHTML = `
      <svg viewBox="0 0 24 24">
        <circle cx="12" cy="12" r="10" />
        <path d="M12 2a10 10 0 0 1 10 10" />
      </svg>
    `;
            this.iframe = document.createElement("iframe");
            this.iframe.className = CSS_CLASSES.iframe;
            this.iframe.setAttribute("allow", "camera; microphone; fullscreen; autoplay; encrypted-media; geolocation");
            this.iframe.setAttribute("title", getTranslations(this.language).ariaLabelModal);
            this.iframe.addEventListener("load", () => this.handleIframeLoad());
            this.container.appendChild(this.loadingEl);
            this.container.appendChild(this.iframe);
            this.embeddedContainer.appendChild(this.container);
        }
        setupEventListeners() {
            this.boundHandleMessage = this.handleMessage.bind(this);
            window.addEventListener("message", this.boundHandleMessage);
            this.boundHandleKeydown = this.handleKeydown.bind(this);
            document.addEventListener("keydown", this.boundHandleKeydown);
        }
        removeEventListeners() {
            if (this.boundHandleMessage) {
                window.removeEventListener("message", this.boundHandleMessage);
                this.boundHandleMessage = null;
            }
            if (this.boundHandleKeydown) {
                document.removeEventListener("keydown", this.boundHandleKeydown);
                this.boundHandleKeydown = null;
            }
        }
        handleMessage(event) {
            if (!isAllowedOrigin(event.origin)) {
                return;
            }
            SDKLogger.log("Received postMessage:", event.data);
            let messageData;
            try {
                if (typeof event.data === "string") {
                    messageData = JSON.parse(event.data);
                }
                else {
                    messageData = event.data;
                }
            }
            catch {
                SDKLogger.warn("Failed to parse postMessage:", event.data);
                return;
            }
            if (messageData.type === "didit:close_request") {
                this.handleCloseRequest();
                return;
            }
            this.callbacks.onMessage(messageData);
        }
        handleKeydown(event) {
            if (!this.state.isOpen)
                return;
            if (event.key === "Escape") {
                event.preventDefault();
                if (this.state.showConfirmation) {
                    this.hideConfirmation();
                }
                else {
                    this.handleCloseRequest();
                }
            }
        }
        handleIframeLoad() {
            if (this.iframe?.src && this.iframe.src !== "about:blank") {
                this.state.isLoading = false;
                this.loadingEl?.classList.add("hidden");
                this.callbacks.onIframeLoad();
            }
        }
        handleCloseRequest() {
            if (this.config.showExitConfirmation) {
                this.showConfirmation();
            }
            else {
                this.callbacks.onCloseConfirmed();
            }
        }
        showConfirmation() {
            this.state.showConfirmation = true;
            this.confirmOverlay?.classList.add("active");
            this.callbacks.onClose();
        }
        hideConfirmation() {
            this.state.showConfirmation = false;
            this.confirmOverlay?.classList.remove("active");
        }
        confirmExit() {
            this.hideConfirmation();
            this.callbacks.onCloseConfirmed();
        }
        open(verificationUrl) {
            this.language = detectLanguageFromUrl(verificationUrl);
            if (!this.overlay && !this.container) {
                this.createDOM();
                this.setupEventListeners();
            }
            SDKLogger.log("Opening with URL:", verificationUrl);
            this.state.isLoading = true;
            this.state.showConfirmation = false;
            this.loadingEl?.classList.remove("hidden");
            this.confirmOverlay?.classList.remove("active");
            if (this.iframe) {
                this.iframe.src = verificationUrl;
            }
            this.state.isOpen = true;
            if (this.embedded) {
                return;
            }
            this.overlay?.classList.add("active");
            document.body.style.overflow = "hidden";
        }
        close() {
            SDKLogger.log("Closing");
            this.state.isOpen = false;
            this.state.isLoading = true;
            this.state.showConfirmation = false;
            if (this.iframe) {
                this.iframe.src = "about:blank";
            }
            if (this.embedded) {
                return;
            }
            this.overlay?.classList.remove("active");
            document.body.style.overflow = "";
        }
        destroy() {
            SDKLogger.log("Destroying");
            this.close();
            this.removeEventListeners();
            if (this.embedded && this.container && this.container.parentNode) {
                this.container.parentNode.removeChild(this.container);
            }
            else if (this.overlay && this.overlay.parentNode) {
                this.overlay.parentNode.removeChild(this.overlay);
            }
            this.overlay = null;
            this.container = null;
            this.iframe = null;
            this.loadingEl = null;
            this.confirmOverlay = null;
        }
        isOpen() {
            return this.state.isOpen;
        }
        isLoading() {
            return this.state.isLoading;
        }
    }

    class DiditSdk {
        static get shared() {
            if (!DiditSdk._instance) {
                DiditSdk._instance = new DiditSdk();
            }
            return DiditSdk._instance;
        }
        static reset() {
            if (DiditSdk._instance) {
                DiditSdk._instance.destroy();
                DiditSdk._instance = null;
            }
        }
        get state() {
            return this._state;
        }
        get configuration() {
            return this._configuration;
        }
        get isPresented() {
            return this._modal?.isOpen() ?? false;
        }
        get errorMessage() {
            return this._errorMessage;
        }
        constructor() {
            this._state = "idle";
            this._modal = null;
            SDKLogger.log("DiditSdk initialized");
        }
        async startVerification(options) {
            const config = options.configuration;
            this._configuration = config;
            SDKLogger.isEnabled = config?.loggingEnabled ?? DEFAULT_CONFIG.loggingEnabled;
            SDKLogger.log("Starting verification with options:", options);
            if (this._modal) {
                this._modal.destroy();
                this._modal = null;
            }
            this._modal = new VerificationModal(config, {
                onClose: () => this.handleModalClose(),
                onCloseConfirmed: () => this.handleModalCloseConfirmed(),
                onMessage: (event) => this.handleVerificationEvent(event),
                onIframeLoad: () => this.handleIframeLoad()
            });
            try {
                const { url } = options;
                if (!url || typeof url !== "string") {
                    throw new Error("Invalid options: url is required");
                }
                this._url = url;
                this.setState("loading");
                this.emitInternalEvent("didit:started", {});
                this._modal?.open(this._url);
            }
            catch (error) {
                this.handleError(error);
            }
        }
        close() {
            SDKLogger.log("Closing verification programmatically");
            this.handleModalCloseConfirmed();
        }
        destroy() {
            SDKLogger.log("Destroying SDK instance");
            this._modal?.destroy();
            this._modal = null;
            this.reset();
        }
        handleModalClose() {
            SDKLogger.log("Modal close requested");
        }
        handleModalCloseConfirmed() {
            SDKLogger.log("Modal close confirmed");
            const sessionData = this.buildSessionData();
            this._modal?.close();
            this.reset();
            const result = {
                type: "cancelled",
                session: sessionData
            };
            this.onComplete?.(result);
        }
        handleIframeLoad() {
            SDKLogger.log("Iframe loaded");
        }
        emitInternalEvent(type, data) {
            const event = {
                type,
                data,
                timestamp: Date.now()
            };
            SDKLogger.log("Emitting internal event:", event);
            this.onEvent?.(event);
        }
        handleVerificationEvent(event) {
            SDKLogger.log("Verification event:", event);
            this.onEvent?.(event);
            switch (event.type) {
                case "didit:ready":
                    SDKLogger.log("Verification iframe ready");
                    break;
                case "didit:started":
                    SDKLogger.log("User started verification");
                    break;
                case "didit:step_started":
                    SDKLogger.log("Step started:", event.data?.step);
                    break;
                case "didit:step_completed":
                    SDKLogger.log("Step completed:", event.data?.step, "-> next:", event.data?.nextStep);
                    break;
                case "didit:media_started":
                    SDKLogger.log("Media started:", event.data?.mediaType, "for step:", event.data?.step);
                    break;
                case "didit:media_captured":
                    SDKLogger.log("Media captured for step:", event.data?.step, "isAuto:", event.data?.isAuto);
                    break;
                case "didit:document_selected":
                    SDKLogger.log("Document selected:", event.data?.documentType, "country:", event.data?.country);
                    break;
                case "didit:verification_submitted":
                    SDKLogger.log("Verification submitted for step:", event.data?.step);
                    break;
                case "didit:code_sent":
                    SDKLogger.log("Code sent via:", event.data?.channel, "codeSize:", event.data?.codeSize);
                    break;
                case "didit:code_verified":
                    SDKLogger.log("Code verified via:", event.data?.channel);
                    break;
                case "didit:status_updated":
                    SDKLogger.log("Status updated:", event.data?.status, "step:", event.data?.step);
                    break;
                case "didit:completed":
                    this.handleVerificationCompleted(event);
                    break;
                case "didit:cancelled":
                    this.handleVerificationCancelled(event);
                    break;
                case "didit:error":
                    this.handleVerificationError(event);
                    break;
                case "didit:step_changed":
                    SDKLogger.log("Step changed:", event.data?.step);
                    break;
            }
        }
        handleVerificationCompleted(event) {
            SDKLogger.log("Verification completed:", event.data);
            const sessionData = this.buildSessionData(event.data);
            if (this._configuration?.closeModalOnComplete) {
                this._modal?.close();
                this.reset();
            }
            const result = {
                type: "completed",
                session: sessionData
            };
            this.onComplete?.(result);
        }
        handleVerificationCancelled(event) {
            SDKLogger.log("Verification cancelled:", event.data);
            const sessionData = this.buildSessionData(event.data);
            this._modal?.close();
            this.reset();
            const result = {
                type: "cancelled",
                session: sessionData
            };
            this.onComplete?.(result);
        }
        handleVerificationError(event) {
            SDKLogger.log("Verification error:", event.data);
        }
        handleError(error) {
            SDKLogger.error("SDK error:", error);
            let verificationError;
            if (error instanceof Error) {
                verificationError = createVerificationError("unknown", error.message);
            }
            else {
                verificationError = createVerificationError("unknown", "An unknown error occurred");
            }
            this._errorMessage = verificationError.message;
            this.setState("error");
            this._modal?.close();
            this.reset();
            const result = {
                type: "failed",
                error: verificationError
            };
            this.onComplete?.(result);
        }
        setState(state) {
            const previousState = this._state;
            this._state = state;
            if (previousState !== state) {
                SDKLogger.log("State changed:", previousState, "->", state);
                this.onStateChange?.(state, this._errorMessage);
            }
        }
        reset() {
            this._state = "idle";
            this._sessionId = undefined;
            this._url = undefined;
            this._errorMessage = undefined;
            this._configuration = undefined;
        }
        buildSessionData(eventData) {
            const sessionId = eventData?.sessionId || this._sessionId;
            if (!sessionId) {
                return undefined;
            }
            return {
                sessionId,
                status: (eventData?.status || "Pending")
            };
        }
    }
    DiditSdk._instance = null;

    exports.DiditSdk = DiditSdk;
    exports.SDK_VERSION = SDK_VERSION;
    exports.default = DiditSdk;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
